autoload/youcompleteme.vim

```diff
@@ -67,6 +67,7 @@ function! youcompleteme#Enable()
   endif
 
   call youcompleteme#EnableCursorMovedAutocommands()
+  call youcompleteme#EnableTextChangedAutocommands()
```

Comment:
Instead of making the the TextChanged autocommands  distinct from what we have currently what do you think about putting the `ifdef` logic inside the already existing `youcompleteme#EnableCursorMovedAutocommands`? One thing to think about is that those autocommands are set with those enable/disable function because other plugins had trouble with them, so instead of breaking again those plugins we could make an effort to maintain backward compatibility. What do you think?